main.go

```diff
@@ -126,6 +173,37 @@ func (c *graphiteCollector) processLines() {
 	}
 }
 
+func parseMetricNameAndTags(name string, labels prometheus.Labels) (string, error) {
+	if strings.ContainsRune(name, ';') {
```

Comment:
I think this could be simpler (and save us from traversing the string twice) if we [unconditionally split the string](https://play.golang.org/p/lUaD6ynuMTm) and iterate over the (potentially zero) tag parts.

Why does this function take labels as an argument? Could it return a fresh `prometheus.Labels` instead?

Reply:
of course. I had it taking labels as an arg since maps are sort of a reference type in go, but making a new set each time and returning it is much cleaning